nengo/utils/progress.py

```diff
@@ -645,35 +647,32 @@ class MultiProgressTracker(object):
         The progress bar to display the progress.
     task : str
         Task name to display.
-    tasks : sequence of str
-        Subtask names to display.
     """
     def __init__(self, progress_bar, task):
-        self.progress_bar = wrap_with_progressupdater(
-            progress_bar=progress_bar)
-        self.task = task
-        self.total_progress = Progress(1, task=self.task)
-        self.subtask = None
-        self.subtask_progress = {}
+        super(MultiProgressTracker, self).__init__(1, progress_bar, task)
 
-    def activate_subtask(self, subtask):
-        self.subtask = subtask
-
-    def subprogress(self, max_steps):
+    def subprogress(self, max_steps, subtask):
         pt = ProgressTracker(
-            max_steps, self.progress_bar, self.subtask, close=False)
-        self.subtask_progress[self.subtask] = pt.progress
+            max_steps, self.progress_bar, subtask, close=False)
+
         return pt
 
+
+class NoopProgressTracker(ProgressTracker):
+    def __init__(self, *_):
+        pass
```

Comment:
Should call the super constructor.

Reply:
I don't think we want to call the super constructor in this case (it'd force us to construct Progress/ProgressBar/ProgressUpdater classes that we never need).

Reply:
I think it is bad practice to derive from a class without calling the constructor (and this caused problems in the Nengo code base before). But I added a commit where this issue becomes irrelevant anyways.